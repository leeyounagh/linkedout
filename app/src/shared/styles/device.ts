const sizes = {
  mobileS: '320px',
  mobileM: '390px',
  mobileL: '425px',
  tablet: '768px',
  laptop: '1024px',
  laptopL: '1440px',
  desktop: '2560px',
  heightS: '480px', 
  heightM: '800px', 
  heightL: '1080px', 
};

export const maxDevices = {
  mobileS: `(max-device-width: ${sizes.mobileS})`,
  mobileM: `(max-device-width: ${sizes.mobileM})`,
  mobileL: `(max-device-width: ${sizes.mobileL})`,
  tablet: `(max-device-width: ${sizes.tablet})`,
  laptop: `(max-device-width: ${sizes.laptop})`,
  laptopL: `(max-device-width: ${sizes.laptopL})`,
  desktop: `(max-device-width: ${sizes.desktop})`,
};

export const minDevices = {
  mobileS: `(min-device-width: ${sizes.mobileS})`,
  mobileM: `(min-device-width: ${sizes.mobileM})`,
  mobileL: `(min-device-width: ${sizes.mobileL})`,
  tablet: `(min-device-width: ${sizes.tablet})`,
  laptop: `(min-device-width: ${sizes.laptop})`,
  laptopL: `(min-device-width: ${sizes.laptopL})`,
  desktop: `(min-device-width: ${sizes.desktop})`,
};

export const maxHeights = {
  heightS: `(max-height: ${sizes.heightS})`,
  heightM: `(max-height: ${sizes.heightM})`,
  heightL: `(max-height: ${sizes.heightL})`,
};

export const minHeights = {
  heightS: `(min-height: ${sizes.heightS})`,
  heightM: `(min-height: ${sizes.heightM})`,
  heightL: `(min-height: ${sizes.heightL})`,
};