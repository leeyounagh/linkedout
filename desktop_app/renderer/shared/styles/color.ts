const color = {
    white: '#FFFFFF',
    black: '#000',
    lightBlack:'#191919',
    pointcolor: '#616FED',
    gray:'#868686',
    gray1:"#686868",
    darkgray:"#121212",
    red:'#E43446'
  }
  
  export default color