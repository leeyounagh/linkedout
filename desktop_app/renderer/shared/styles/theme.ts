import { DefaultTheme } from 'styled-components';


export const lightTheme: DefaultTheme = {
    colors: {
      background: '#ffffff',
      text: '#000000',
    },
  };
  
  export const darkTheme: DefaultTheme = {
    colors: {
      background: ' #0E0E0E',
      text: '#ffffff',
    },
  };