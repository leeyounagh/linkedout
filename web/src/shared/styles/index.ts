export { Devices } from './device';
export { lightTheme, darkTheme } from './theme';