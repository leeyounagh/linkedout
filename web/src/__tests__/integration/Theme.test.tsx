import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { describe } from "node:test";
import RootLayout from "@/app/layout";
import { lightTheme, darkTheme } from "@/shared/styles/index";

describe("다크모드 테마적용 테스트", () => {
  it("다크모드 버튼 클릭", () => {
    const { container } = render(
      <RootLayout>
        <button>다크모드 버튼</button>
      </RootLayout>
    );
    const button = screen.getByText("다크모드 버튼");

    // 초기 테마가 다크 테마인지 확인
    expect(container.firstChild).toMatchSnapshot("Dark Mode Initial");
    expect(document.body).toHaveStyle({
      backgroundColor: darkTheme.background,
      color: darkTheme.color,
    });

    // 버튼 클릭하여 라이트 모드로 전환
    fireEvent.click(button);
    expect(container.firstChild).toMatchSnapshot("Light Mode After Click");
    expect(document.body).toHaveStyle({
      backgroundColor: lightTheme.background,
      color: lightTheme.color,
    });

    // 버튼을 다시 클릭하여 다크 모드로 전환
    fireEvent.click(button);
    expect(container.firstChild).toMatchSnapshot(
      "Dark Mode After Second Click"
    );
    expect(document.body).toHaveStyle({
      backgroundColor: darkTheme.background,
      color: darkTheme.color,
    });
  });
});
