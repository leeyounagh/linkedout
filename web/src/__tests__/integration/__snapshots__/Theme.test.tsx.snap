// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`다크모드 버튼 클릭: Dark Mode After Second Click 1`] = `
<html
  lang="en"
>
  <head>
    <link
      href="/icon-192x192.png"
      rel="icon"
      sizes="192x192"
    />
    <link
      href="/icon-256x256.png"
      rel="icon"
      sizes="256x256"
    />
    <link
      href="/icon-384x384.png"
      rel="icon"
      sizes="384x384"
    />
    <link
      href="/icon-512x512.png"
      rel="icon"
      sizes="512x512"
    />
  </head>
  <body>
    <div
      class="container"
    >
      <button>
        다크모드 버튼
      </button>
    </div>
  </body>
</html>
`;

exports[`다크모드 버튼 클릭: Dark Mode Initial 1`] = `
<html
  lang="en"
>
  <head>
    <link
      href="/icon-192x192.png"
      rel="icon"
      sizes="192x192"
    />
    <link
      href="/icon-256x256.png"
      rel="icon"
      sizes="256x256"
    />
    <link
      href="/icon-384x384.png"
      rel="icon"
      sizes="384x384"
    />
    <link
      href="/icon-512x512.png"
      rel="icon"
      sizes="512x512"
    />
  </head>
  <body>
    <div
      class="container"
    >
      <button>
        다크모드 버튼
      </button>
    </div>
  </body>
</html>
`;

exports[`다크모드 버튼 클릭: Light Mode After Click 1`] = `
<html
  lang="en"
>
  <head>
    <link
      href="/icon-192x192.png"
      rel="icon"
      sizes="192x192"
    />
    <link
      href="/icon-256x256.png"
      rel="icon"
      sizes="256x256"
    />
    <link
      href="/icon-384x384.png"
      rel="icon"
      sizes="384x384"
    />
    <link
      href="/icon-512x512.png"
      rel="icon"
      sizes="512x512"
    />
  </head>
  <body>
    <div
      class="container"
    >
      <button>
        다크모드 버튼
      </button>
    </div>
  </body>
</html>
`;
